Basic/type_traits: use `__is_identifier` for some feature checks

MaxDesiatov opened the pull request:
When building on Alpine with Musl `__has_feature(is_trivially_constructible)` does not work, even though `__is_trivially_constructible` is defined and `__has_trivial_constructor` is deprecated. Same for `__is_trivially_destructible` and `__has_trivial_destructor` respectively.

MaxDesiatov committed:
Basic/type_traits: use `__is_identifier` for some feature checks

When building on Alpine with Musl `__has_feature(is_trivially_constructible)` does not work, even though `__is_trivially_constructible` is defined and `__has_trivial_constructor` is deprecated. Same for `__is_trivially_destructible` and `__has_trivial_destructor` respectively.

MaxDesiatov commented:
@swift-ci smoke test

al45tair reviewed (approved):
Seems reasonable enough; is it worth _also_ testing using `__has_feature()` in case your `__has_keyword()` macro fails for some reason on a system that does have the feature but isn't using libc++ or MSVC?

MaxDesiatov committed:
Basic/type_traits: use `__has_feature` or `__has_keyword`

That potentially makes these checks more future-proof in case more edge cases are introduced on different platforms.

MaxDesiatov commented:
@swift-ci smoke test
